### ISSUE TITLE
Tree-sitter syntax highlighting for JS is not applied consistently with certain themes

### ISSUE THREAD

**Issue by eeshugerman** [syl20bnr/spacemacs#15356]
#### Description :octocat:
When the `tree-sitter` layer is installed and `tree-sitter-syntax-highlight-enable` is `t`, tree-sitter syntax highlighting is not applied consistently in JavaScript files. Sometimes part (and sometimes all?) of the buffer appears to be highlighted with the regex-based system instead. `revert-buffer` seems to fix it, at least temporarily for part of the buffer currently visible.

In the screenshot below, we can see tree-sitter highlighting is working properly for first ~60% of lines, but then halfway through the `Constants.ModelType.PROJECT` symbol the highlighting changes to the regex-based style.

![tree-sitter-spaz](https://user-images.githubusercontent.com/25518211/154525562-e677c51f-fcab-48cc-b205-11d1bd98b43a.png)

I haven't yet determined if this is present in a vanilla Emacs+tree-sitter config, or vanilla Spacemacs+tree-sitter config, or if it's something specific to my config. 

#### Reproduction guide :beetle:
- Start Emacs
- Open a large JS file
- Browse around, edit some stuff

*Observed behaviour:* :eyes: :broken_heart:
Often (when?) syntax highlighting becomes inconsistent -- a mix of tree-sitter-based and regex-based styles. 

*Expected behaviour:* :heart: :smile:
Tree-sitter-based syntax highlighting is applied consistently.

#### System Info :computer:
- OS: darwin
- Emacs: 28.0.60
- Spacemacs: 0.999.0
- Spacemacs branch: develop (rev. 0bdddfc16)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(graphviz ansible auto-completion c-c++ csv dap debug docker emacs-lisp epub erc git github groovy haskell helpful html ibuffer ipython-notebook ivy java javascript markdown multiple-cursors nav-flash nginx org posframe prettier python ruby rust scheme shell shell-scripts slack spacemacs-layouts spell-checking sql syntax-checking terraform tree-sitter treemacs typescript unicode-fonts yaml)
```
- System configuration features: ACL GIF GLIB GMP GNUTLS JPEG JSON LCMS2 LIBXML2 MODULES NATIVE_COMP NOTIFY KQUEUE NS PDUMPER PNG RSVG THREADS TIFF TOOLKIT_SCROLL_BARS XIM ZLIB

#### Backtrace :paw_prints:
N/A

**Comment by eeshugerman** [comment]
A recent package update seems to have fixed this for me 🤔

**Comment by eeshugerman** [comment]
Sorry for the switcheroo @lebensterben .

I just realized that it wasn't a package update that fixed it, it was switching themes: I'm only seeing this issue with `spacemacs-dark` and `spacemacs-light`. The rest that I tried (various `doom-` themes, both `modus-` themes) do not exhibit the issue.

**Comment by lebensterben** [comment]
this doesn't make any sense...
a theme only defines the font faces various text uses.

**Comment by lebensterben** [comment]
You may want to report this to the upstream if you believe it's caused by the theme. https://github.com/nashamri/spacemacs-theme

**Comment by eeshugerman** [comment]
I think there might be some sort of race condition with `jit-lock` (??): I've found that increasing `jit-lock-chunk-size` from the default 1500 to 5000 fixes the problem in a 200 line buffer, and setting `jit-lock-defer-time` to 0.2 makes it much worse (as if tree-sitter were disabled entirely). I notice the spacemacs themes are much larger than the other themes I tried, maybe that could have something to do with it. Good point, I'll report to both upstreams once I have a chance to explore a little more.

**Comment by mohammedzeglam-pg** [comment]
the problem is simple no queries in build directory that is what happened with me and not supporting tsx files.
tree-sitter depends on s-expression to do highlight but no file found

**Comment by lebensterben** [comment]
@mohammedzeglam-pg 

what do you mean? can you rephrase and explain it?

**Comment by eeshugerman** [comment]
This seems to be fixed, presumably by a package update. Anyone feel free to reopen if you run into this.
